test_db.py

```diff
@@ -0,0 +1,231 @@
+import io
+import base64
+import db.queries.user as user
+import db.queries.seed as seed
+import db.queries.picture as picture
+import metadata.picture_set as picture_set_data
+import metadata.picture as picture_data
+import db.__init__ as db
+
+"""
+This is a test script for the database packages. 
+It tests the functions in the user, seed and picture modules.
+"""
+
+
+def is_valid_uuid(val):
+    """
+    This validates if a given string is a UUID
+    """
+    try:
+        uuid.UUID(str(val))
+        return True
+    except ValueError:
+        return False
+
+
+# --------------------  USER FUNCTIONS --------------------
+class test_user_functions(unittest.TestCase):
+    def setUp(self):
+        self.con = db.connect_db()
+        self.cursor = self.con.cursor()
+        self.email = "test@email.gouv.ca"
+
+    def tearDown(self):
+        db.end_query(self.con, self.cursor)
+
+    def test_is_user_registered(self):
+        """
+        This test checks if the is_user_registered function returns the correct value
+        for a user that is not yet registered and one that is.
+        """
+
+        self.assertFalse(
+            user.is_user_registered(self.cursor, self.email),
+            "The user should not already be registered",
+        )
+
+        user_id = user.register_user(self.cursor, self.email)
+
+        self.assertTrue(is_valid_uuid(user_id), "The user_id is not a valid UUID")
+        self.con.rollback()
+
+    def test_get_user_id(self):
+        """
+        This test checks if the get_user_id function returns the correct UUID
+        """
+        user_id = user.register_user(self.cursor, self.email)
+        uuid = user.get_user_id(self.cursor, self.email)
+
+        self.assertTrue(is_valid_uuid(user_id), "The user_id is not a valid UUID")
+        self.assertTrue(is_valid_uuid(uuid), "The returned UUID is not a valid UUID")
+        self.assertEqual(
+            user_id, uuid, "The returned UUID is not the same as the registered UUID"
+        )
+
+        self.con.rollback()
+
+    def test_register_user(self):
+        """
+        This test checks if the register_user function returns a valid UUID
+        """
+        user_id = user.register_user(self.cursor, self.email)
+
+        self.assertTrue(is_valid_uuid(user_id), "The user_id is not a valid UUID")
+
+        self.con.rollback()
+
+
+# --------------------  SEED FUNCTIONS --------------------
+
+
+class test_seed_functions(unittest.TestCase):
+    def setUp(self):
+        self.con = db.connect_db()
+        self.cursor = self.con.cursor()
+        self.seed_name = "test"
+
+    def tearDown(self):
+        db.end_query(self.con, self.cursor)
+
+    def test_get_all_seeds_names(self):
+        """
+        This test checks if the get_all_seeds_names function returns a list of seeds
+        with at least the one seed we added
+        """
+        seed.new_seed(self.cursor, self.seed_name)
+        seeds = seed.get_all_seeds_names(self.cursor)
```

Comment:
In this test, I think it would be more relevant to have a set of expected seed_name and to test if the name return by the function are the same. To do that without inserting the test name in the database, you can use the `@patch` decorator to mock the return value of a function.

Reply:
I dont want this test to expect the tables to be populated. This is why I insert a value. I will add more than one value because I should test out it really returns a list and not only one seed.